CRM/Contribute/Form/Task/PDFLetterCommon.php

```diff
@@ -228,6 +228,8 @@ private static function resolveTokens($html_message, $contact, $contribution, $m
       // no change to normal behaviour to avoid risk of breakage
       $tokenHtml = CRM_Utils_Token::replaceContributionTokens($tokenHtml, $contribution, TRUE, $messageToken);
     }
+    $domain = CRM_Core_BAO_Domain::getDomain();
+    $tokenHtml = CRM_Utils_Token::replaceDomainTokens($tokenHtml, $domain, TRUE, $messageToken);
```

Comment:
@seamuslee001 Can you put this above `replaceContactTokens()` - we found a scenario with the mailings recently when for some reason `replaceContactTokens()` stripped the domain tokens and they didn't get set.